gnomad_qc/v4/create_release/create_false_dup_liftover.py

```diff
@@ -0,0 +1,176 @@
+    return (
+        f"{_release_root(version=release_version, test=test, data_type=data_type)}/gnomad.{data_type}.v{release_version}_three_false_dup_genes_liftover.ht"
+    )
+
+
+def filter_liftover_to_false_dups(
+    data_type: str,
+) -> hl.Table:
+    """
+    Read in gnomAD v2 liftover table, filter to genes of interest, and return the Table.
+
+    :param data_type: String of either "exomes" or "genomes" for data type.
+    :return: Filtered Hail Table.
+    """
+    ht = hl.read_table(
+        get_gnomad_liftover_data_path(data_type=data_type, version="2.1.1")
+    )
+
+    # Filter to chr21, since it is known that is where all 3 of the genes are.
+    ht = ht.filter(ht.locus.contig == "chr21")
+
+    # Filter to any variant located in one of the 3 genes of interest.
+    ht = ht.filter(
+        hl.set(FALSE_DUP_GENES).any(
+            lambda gene_symbol: ht.vep.transcript_consequences.gene_symbol.contains(
+                gene_symbol
+            )
+        )
+    )
+
+    return ht
+
+
+def main(args):
+    """
+    There are three clinically relevant genes impacted by false duplications in the GRCh38 reference (KCNE1, CBS, and CRYAA).
+    To make it easier for our users to switch to gnomAD v4 and the new reference build,
+    this script creates a new release file that combines information from the v2 exomes and genomes only for these three genes.
+    To create this file, filter the v2 liftover files (https://gnomad.broadinstitute.org/downloads#v2-liftover) to these genes,
+    and then merge frequency information across the exomes and genomes
+    """
+    # Read in liftover files.
+    exome_ht = filter_liftover_to_false_dups("exomes")
+    genome_ht = filter_liftover_to_false_dups("genomes")
+
+    # Union and merge to contain any variant present in either.
+    # Note: only about ~550 overlap, with ~187k exome variants and ~12k genome variants.
+    # This makes sense in exonic regions.
+    # Genome global annotations stored with _1
+    ht = exome_ht.select().join(genome_ht.select(), how="outer")
+
+    # Annotate with information from both exomes and genomes - many will be NaN.
+    ht = ht.annotate(
+        v2_exomes=exome_ht[ht.locus, ht.alleles],
+        v2_genomes=genome_ht[ht.locus, ht.alleles],
+    )
+
+    # Checkpoint output to new temp file.
+    ht = ht.checkpoint(hl.utils.new_temp_file("three_dup_genes_liftover", "ht"))
+
+    # Merge exomes and genomes frequencies.
+    # Code adapted from
+    # gnomad_qc/v4/create_release/create_combined_faf_release_ht.py#L155
+    # as of 01-26-2025
+    # Changed for 'gen_anc' -> pop
+    freq, freq_meta = merge_freq_arrays(
+        farrays=[ht.v2_exomes.freq, ht.v2_genomes.freq],
+        fmeta=[ht.index_globals().freq_meta, ht.index_globals().freq_meta_1],
+    )
+
+    # Select to first 28 groups, containing:
+    # Raw, adj, adj male & female, and adj sex-split pops for 8 pops.
+    # Hail sets this to first 28 of each entry in the array, not first 28 of array.
```

Comment:
unclear what this comment is referring to

Reply:
freq is an array, and each element of the array has a struct (struct{AC: int32, AF: float64, AN: int32, homozygote_count: int32}>), freq[:28] is pulling out the first 28 elements of the array